### DISCUSSION TITLE
Converting EPS file yields an error

### DISCUSSION THREAD

**Discussion by IsaacTFM** [ImageMagick/ImageMagick#4137] [Help] [answered]
Hello! Whenever I try to convert an EPS file, I am greeted with this error:
`convert: no images defined 'image_new.png' @ error/convert.c/ConvertImageCommand/3322`
Here is the command I used:
`magick convert image.eps image_new.png`
Yes, image.eps exists, yes, it is in the current working dictionary.

**Comment by fmw42** [comment]
In IM 7, use magick not magick convert.  If that does not help, then post your EPS file. Also tell us what your IM version and platform are!

**Reply by IsaacTFM** [reply]
> In IM 7, use magick not magick convert. If that does not help, then post your EPS file. Also tell us what your IM version and platform are!

It's Windows, and it gives this full error when I use magick:

`magick: unable to open image 'magick': No such file or directory @ error/blob.c/OpenBlob/3536.
magick: no decode delegate for this image format '' @ error/constitute.c/ReadImage/572.`

**Comment by snibgo** [comment]
@IsaacTFM: Please also say whether that is the _complete_ error message. If it isn't, please post the complete text.

**Reply by IsaacTFM** [reply]
Here is the full error:
`
convert: FailedToExecuteCommand '"gswin64c.exe" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pngalpha" -dTextAlphaBits=4 -dGraphicsAlphaBits=4 "-r72x72" -g561x471 -dEPSCrop  "-sOutputFile=C:/Users/USER/AppData/Local/Temp/magick-YJ861AegCI4LlqMlKaqCKBr0RfHIGs_F%d" "-fC:/Users/USER/AppData/Local/Temp/magick-LtFxmFlGu9dfrNC3Fpamr6402Y6U1opv" "-fC:/Users/USER/AppData/Local/Temp/magick-2f8mti_7rASJm1UAMAWt2TQJO6GFw0c9"' (The system cannot find the file specified.
) @ error/delegate.c/ExternalDelegateCommand/518.
convert: FailedToExecuteCommand '"gswin64c.exe" -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pngalpha" -dTextAlphaBits=4 -dGraphicsAlphaBits=4 "-r72x72" -g561x471 -dEPSCrop  "-sOutputFile=C:/Users/USER/AppData/Local/Temp/magick-YJ861AegCI4LlqMlKaqCKBr0RfHIGs_F%d" "-fC:/Users/USER/AppData/Local/Temp/magick-LtFxmFlGu9dfrNC3Fpamr6402Y6U1opv" "-fC:/Users/USER/AppData/Local/Temp/magick-2f8mti_7rASJm1UAMAWt2TQJO6GFw0c9" -c showpage' (The system cannot find the file specified.
) @ error/delegate.c/ExternalDelegateCommand/518.
convert: Postscript delegate failed 'The system cannot find the file specified.
' @ error/ps.c/ReadPSImage/843.
convert: no images defined 'image_new.png' @ error/convert.c/ConvertImageCommand/3322.
`

**Comment by IsaacTFM** [comment]
Do I need to install Ghostscript to use EPS files?

**Comment by fmw42** [comment, marked as answer]
Use magick.exe on windows, typically.

What do you get from 

`magick.exe -version`

You do need Ghostscript installed so that ImageMagick can find it. Does it list gs or gslib in the list of delegates?

**Comment by snibgo** [comment]
Whenever you get error messages from anything, the _first_ message is often the important one. Your is:
```
convert: FailedToExecuteCommand '"gswin64c.exe" 
:
(The system cannot find the file specified. )
```
So, IM is trying to run gswin64c.exe, which is Ghostscript. Install Ghostscript and try again.

**Comment by IsaacTFM** [comment]
Okay, thanks for the help. Due to the security flaws of Ghostscript, this appears to be a route I don't wish to follow. Thanks all the same!

**Reply by fmw42** [reply]
Those security flaws have been fixed a long time ago. Current versions are fine.  But you may have to edit your ImageMagick policy.xml file to enable the use of PDF/PS/EPS files once you have Ghostscript installed.
